mica/archive/aca_hdr3.py

```diff
@@ -23,12 +23,21 @@
 
 
 def two_byte_sum(byte_msids, scale=1):
-    def func(slot_data):
-        return (
-            (slot_data[byte_msids[0]].astype("int") >> 7) * (-1 * 65535)
-            + (slot_data[byte_msids[0]].astype("int") << 8)
-            + (slot_data[byte_msids[1]].astype("int"))
-        ) * scale
+    def func(slot_data) -> np.ma.MaskedArray:
+        # For each pair bytes0[i], bytes1[i], return the 16-bit signed integer
+        # corresponding to those two bytes. The input bytes are unsigned.
+        bytes0 = slot_data[byte_msids[0]].astype(np.uint8)
+        bytes1 = slot_data[byte_msids[1]].astype(np.uint8)
+
+        # Make a 2xN array, then transpose to Nx2, then flatten to 2N, then copy to
+        # get values continous in memory.
+        bytes8_2xN = np.ma.vstack([bytes0, bytes1], dtype=np.uint8)
+        bytes8 = bytes8_2xN.transpose().flatten().copy()
+
+        # Now view the 2N bytes as N 16-bit signed integers.
+        ints16 = np.ma.array(bytes8.data.view(">i2"), mask=bytes8.mask[::2])
```

Comment:
Thanks for the fix!  Sorry I was not paying enough attention to the way to transform bytes to ints in general.  I do wonder, now that I'm paying more attention, if instead of ending up with transpose().flatten().copy() + view if an updated shift and cast would end up being more readable
```
ints16 = ((bytes0.astype(np.uint16) << 8) | bytes1).astype(np.int16)
```
What are your thoughts?

Reply:
What you wrote might be more concise and (with more investigation and testing) might work. But this PR is already so far down in the weeds, let's just go with the already-tested version.